trail/upload/models.py

```diff
@@ -3,29 +3,49 @@
 """Define the SQL metadata design here."""
 
 
-class ExampleFrame(models.Model):
-    # meta-class mixin ??
-    class Meta:
-        # Django doesn't do composite primary keys
-        constraints = [
-            models.UniqueConstraint(fields=['run', 'camcol', 'filter', 'field'],
-                                    name="SDSS unique identifiers.")
-        ]
-
-    run = models.IntegerField()
-    camcol = models.IntegerField()
-    filter = models.CharField(max_length=1)
-    field = models.IntegerField()
-
-    ctype = models.CharField(max_length=20, null=False)
-    crpix1 = models.FloatField(null=False)
-    crpix2 = models.FloatField(null=False)
-    crval1 = models.FloatField(null=False)
-    crval2 = models.FloatField(null=False)
-
-    cd11 = models.FloatField(null=False)
-    cd12 = models.FloatField(null=False)
-    cd21 = models.FloatField(null=False)
-    cd22 = models.FloatField(null=False)
-
-    t = models.DateTimeField(verbose_name="UTC time at exposure start.")
+__all__ = ["UploadInfo", "Metadata", "Wcs"]
+
+
+class UploadInfo(models.Model):
+    # integer autoincrement primary fields is automatically added by Django
+
+    created = models.DateTimeField("date on which data was uploaded", auto_now_add=True)
+    ip = models.GenericIPAddressField("IP address of the uploader", default="127.0.0.1")
+
+
+class Metadata(models.Model):
+
+    # This will need to be fixed, cascading can orphan medata entries
```

Comment:
This is me struggling against the Django ORM and schema layouts. Technically I think `CASCADING` would be fine here too, because it works the other way around, but I'm not sure how Django deals with orphaning in reverse so I had these protected. 

Of course, with how we are processing (in steps) we will need to address how we rollback changes and deal with cleanup. This however made thinking about the problem, for me, much harder so I operated under the assumption that all data we get will not fail just to get the code going. 

I think handling failures (both the very explicit ones like raised errors but also more subtle processing errors like badly processed data, or correct data but someone is trolling us with not-science images in data arrays) is really important but it's also confusing. This deserves an issue of its own where rollback and validation behavior is precisely defined. 

A part of the reason why this is so hard is that I'm not even sure what are all the failure cases. Now that we have something, this should definitely get tested out in anger and failures mapped out so that we know what our edge cases are. Maybe someone cleverer than me can reason about it in advance - in which case `FitsProcessor.process` method would be the most appropriate place to start implementing rollback behavior.

For most of the code in this PR I have assumed that no uploaded data will break the code in any non-trivial ways.

Reply:
Hmmm... let's discuss more later, because I'd like to better understand the subtleties here. I think this is fine for now, just want to make sure we flag it to revisit when friendly data can't be assumed.

Reply:
I swear I wrote a reply and GitHub ate it. Let's discuss more, because I'd like to understand the subtleties here better. For now, I'm OK with assuming uploaded data is friendly, but we'll need to break that assumption eventually.

Reply:
Weird, I see both replies. Ok, issue https://github.com/dirac-institute/trailblazer/issues/15 - this will be added in a separate issue (as much as possible) and we will discuss details at a later date. I will mark discussions like these as resolved here.